File: 1319-unique-number-of-occurrences/unique-number-of-occurrences.cpp
class Solution {
public:

    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int, int> val2freq;
        for (int val : arr)
          val2freq[val]++;
        
        unordered_set<int> freq_set;
        for (auto& it : val2freq)
        {
          if (!freq_set.count(it.second))
            freq_set.insert(it.second);
          else
            return false;
        }
        return true;
    }
/*
    bool uniqueOccurrences(vector<int>& arr) {
        unordered_map<int, int> val2freq;
        for (int val : arr)
          val2freq[val]++;
        
        unordered_map<int, int> freq2occur;
        for (auto& it : val2freq)
        {
          freq2occur[it.second]++;
          if (freq2occur[it.second] > 1)
            return false;
        }
        
        return true;
    }
*/
};

File: 1319-unique-number-of-occurrences/unique-number-of-occurrences.java
class Solution {
    public boolean uniqueOccurrences(int[] arr) {
        Map<Integer, Integer> val2freq = new HashMap();
        for (int val: arr)
          val2freq.put(val, val2freq.getOrDefault(val, 0) + 1);
    
        Set<Integer> freq_set = new HashSet();
        for (int freq: val2freq.values())
        {
            if (!freq_set.add(freq))
              return false;
        }

        return true;
    }
}